rasa/graph_components/converters/nlu_message_converter.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, Optional, List
+
+from rasa.core.channels.channel import UserMessage
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.nlu.constants import TEXT, INTENT, ENTITIES
+from rasa.shared.nlu.training_data.message import Message
+
+
+class NLUMessageConverter(GraphComponent):
+    """Converts the user message into a NLU Message object."""
+
+    def __init__(self, config: Dict[Text, Any]) -> None:
+        """Creates converter from config."""
+        self._config = config
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns default configuration (see parent class for full docstring)."""
+        return {
+            "remove_duplicates": True,
+            "unique_last_num_states": None,
+            "augmentation_factor": 50,
+            "tracker_limit": None,
+            "use_story_concatenation": True,
+            "debug_plots": False,
+        }
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> NLUMessageConverter:
+        """Creates component (see parent class for full docstring)."""
+        return cls(config)
+
+    @staticmethod
+    def convert_user_message(message: Optional[UserMessage]) -> List[Optional[Message]]:
+        """Converts user message into Message object.
+
+        Returns:
+            List containing only one instance of Message.
+            Else empty list if user message is None.
+        """
+        if message:
+            data = dict()
+            data[TEXT] = message.text
+            data["input_channel"] = message.input_channel
```

Comment:
Do we actually need the input channel here? [This code](https://github.com/RasaHQ/rasa/blob/f30fade97c9fd50b7156aa7d8513bc2fc3527a78/rasa/core/processor.py#L589-L591) is currently not doing that. From my intuition it's fine if the [`NLUPredictionToHistoryAdder`](https://github.com/RasaHQ/rasa/issues/9327) takes care of passing the `input_channel` to the `UserUttered` event.

Reply:
I wasn't sure how to check what does `Message.data` contain to be sure if `input_channel` is relevant or not 🤔

Reply:
I'd check the line in the existing code where `Message` is created. There it doesn't add the `input_channel` to `Message`)

Reply:
> I'd check the line in the existing code where Message is created. 

Are you referring to this [line](https://github.com/RasaHQ/rasa/blob/7f0bb29b1a40e237931f6f431813e08654bea1fb/rasa/nlu/model.py#L471)? 
Ok, yup, will drop the `input_channel`.

Reply:
yep exactly 👍🏻